nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java

```diff
@@ -806,7 +806,22 @@ private void setReceiverNonnull(
   @Override
   public TransferResult<Nullness, NullnessStore> visitAssertionError(
       AssertionErrorNode assertionErrorNode, TransferInput<Nullness, NullnessStore> input) {
-    return noStoreChanges(NULLABLE, input);
+
+    Node condition = assertionErrorNode.getCondition();
+
+    if (condition == null
+        || !(condition instanceof NotEqualNode)
+        || !(((NotEqualNode) condition).getRightOperand() instanceof NullLiteralNode)
+        || !(((NotEqualNode) condition).getLeftOperand() instanceof FieldAccessNode)) {
+      return noStoreChanges(NULLABLE, input);
+    }
+
+    AccessPath accessPath =
```

Comment:
You need to do a null check on `accessPath`, as there are some field access expressions that we cannot currently express with an `AccessPath`

Reply:
Actually why just field accesses here and not also variables and method calls? We should handle any expression we can represent with an `AccessPath`